File: ohgod/src/main/java/org/codecop/rogue/template/Room.java
package org.codecop.rogue.template;

public class Room {

    public static final char PLAYER = '@';
    public static final char FLOOR = ' ';
    private char[][] baseLayout;

    public Room() {
        this.baseLayout = getBaseLayout();
        this.baseLayout[1][1] = PLAYER;
    }

    private static char[][] getBaseLayout() {
        return new char[][]{
                "#######\n".toCharArray(),
                "#     #\n".toCharArray(),
                "|     #\n".toCharArray(),
                "#     #\n".toCharArray(),
                "#######\n".toCharArray()
        };
    }

    @Override
    public String toString() {
        String map = "";
        for(char[] line: this.baseLayout) {
            map = map.concat(String.copyValueOf(line));
        }
        return map;
    }

    public void setNewPosition(int row, int column) {
        this.baseLayout = getBaseLayout();
        this.baseLayout[row][column] = PLAYER;
    }
}

File: BRUTE_FORCE/src/Room.ts
export default class Room {
  player = "@";
  returnLayout(): string {
    const layout = "######\n#    #\n#    |\n#    #\n######";
    // console.log(layout);
    return layout;
  }

  setNewPosition(row: number, column: number): string {
    const layout = this.returnLayout()
      .split("\n")
      .map((line) => line.split(""));
    layout[row][column] = this.player;
    const newLayout = layout.map((line) => line.join("")).join("\n");
    return newLayout;
  }
}
